operator/internal/controller/paladinregistration_controller.go

```diff
@@ -64,7 +64,8 @@ var registryABI = abi.ABI{
 // PaladinRegistrationReconciler reconciles a PaladinRegistration object
 type PaladinRegistrationReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	Scheme   *runtime.Scheme
+	RPCCache *rpcCache
```

Comment:
nit: `RPCClientManager` ? Cache is a bit odd term for a map of HTTP clients, and implies expiry/LRU/etc which isn't implemented (yet).